packages/lib/src/data-grid/DataGrid.tsx

```diff
@@ -0,0 +1,329 @@
+                        });
+                        return newRows;
+                      });
+                    } else {
+                      const rowIndex = rowsToRender.findIndex((rowToRender) => row === rowToRender);
+                      setRowsToRender((rows) => {
+                        const newRows = [...rows];
+                        deleteRow(newRows, rowIndex + 1);
+                        return newRows;
+                      });
+                    }
+                  }}
+                />
+              </ActionContainer>
+            );
+          },
+        },
+        ...expectedColumns,
+      ];
+    }
+    if (!expandable && rows.some((row) => Array.isArray(row.childRows) && row.childRows.length > 0) && uniqueRowId) {
+      // only the first column will be clickable and will expand the rows
+      const firstColumnKey = expectedColumns[0].key;
+      expectedColumns[0] = {
+        ...expectedColumns[0],
+        renderCell({ row }) {
+          if (row.childRows?.length) {
+            return (
+              <HierarchyContainer level={row.rowLevel || 0}>
+                {renderHierarchyTrigger(rowsToRender, row, uniqueRowId, firstColumnKey, (rows) =>
+                  setRowsToRender(rows)
+                )}
+              </HierarchyContainer>
+            );
+          }
+          return (
+            <HierarchyContainer level={row.rowLevel || 0} className="ellipsis-cell">
+              {row[firstColumnKey]}
+            </HierarchyContainer>
+          );
+        },
+      };
+    }
+    if (selectable) {
+      expectedColumns = [
+        {
+          key: "selected",
+          name: "",
+          maxWidth: 50,
+          width: 50,
+          renderCell({ row }) {
+            if (!row.isExpandedChildContent) {
+              return (
+                <ActionContainer>{renderCheckbox(rows, row, uniqueRowId, selectedRows, onSelectRows)}</ActionContainer>
+              );
+            }
+          },
+          renderHeaderCell: () => {
+            return (
+              <ActionContainer>{renderHeaderCheckbox(rows, uniqueRowId, selectedRows, onSelectRows)}</ActionContainer>
+            );
+          },
+        },
+        ...expectedColumns,
+      ];
+    }
+    return expectedColumns;
+  }, [selectable, expandable, columns, rowsToRender, onSelectRows, rows, summaryRow, uniqueRowId, selectedRows]);
+  // array with the order of the columns
+  const [columnsOrder, setColumnsOrder] = useState((): number[] => columnsToRender.map((_, index) => index));
+  const [sortColumns, setSortColumns] = useState<readonly SortColumn[]>([]);
+  // const [visibleColumns, setVisibleColumns] = useState(
+  //   columnsNamesIntoOptions(columns).map((visibleColumn) => {
+  //     return visibleColumn.value;
+  //   })
+  // );
+
+  useEffect(() => {
+    setColumnsOrder(columnsToRender.map((_, index) => index));
+  }, [columnsToRender]);
+
+  const reorderedColumns = useMemo(() => {
```

Comment:
Is it really needed to have a separate object for `reorderedColumns` instead of applying a sort to the `columnsToRender` themselves?

Reply:
I find it easier to understand this way, by using a useMemo we have it isolated to the rest of the functionality and it is also recalculated if the order changes.